frontend/src/components/general/fileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,155 @@
+                onChange={handleFileChange}
+                className="hidden"
+            />
+
+            {/* Button to trigger the file input dialog */}
+            {!selectedFile && (
+                <button
+                    className="w-80 h-36 text-lg font-medium flex flex-col items-center justify-center gap-4 text-indigo-600 bg-black border-2 border-dashed border-indigo-600 rounded-xl cursor-pointer transition duration-300 hover:bg-gray-100"
+                    onClick={onChooseFile}
+                >
+                    <span className="w-12 h-12 text-4xl text-indigo-600 flex items-center justify-center rounded-full bg-indigo-50">
+                        ⬆️
+                    </span>
+                    Datei hochladen
+                </button>
+            )}
+
+            {selectedFile && (
+                <>
+                    <div className="w-72 p-4 bg-black rounded-lg flex items-center gap-4 border border-indigo-300">
+                        <span className="text-indigo-600 text-2xl">📄</span>
+
+                        <div className="flex-1">
+                            <h6 className="text-sm font-medium">{selectedFile?.name}</h6>
+
+                            <div className="w-full h-1.5 bg-gray-200 rounded mt-2">
+                                <div
+                                    className="h-1.5 bg-indigo-600 rounded"
+                                    style={{ width: `${progress}%` }}
+                                />
+                            </div>
+                        </div>
+
+                        {uploadStatus === status.select ? (
+                            <button
+                                className="w-9 h-9 text-indigo-600 bg-indigo-50 rounded-full flex items-center justify-center"
+                                onClick={clearFileInput}
+                            >
+                                <Trash2 />
+                            </button>
+                        ) : (
+                            <div className="w-9 h-9 bg-indigo-50 rounded-full flex items-center justify-center text-indigo-600">
+                                {uploadStatus === "uploading" ? (
+                                    `${progress}%`
+                                ) : uploadStatus === status.done ? (
+                                    <button
+                                    className="w-9 h-9 text-indigo-600 bg-indigo-50 rounded-full flex items-center justify-center"
+                                    onClick={clearFileInput}
+                                >
+                                    <Check />
+                                </button>
+                                ) : null}
+                            </div>
+                        )}
+                    </div>
+                    <br />
+                    <button onClick={handleUpload} className="w-72 p-4 bg-green-dark text-white px-5 py-2 group inline-flex gap-x-3 
```

Comment:
I can’t use the primaryButton directly since the button label changes dynamically based on the upload status. When a file is selected, the label displays 'Upload File', and when no file is selected, it shows 'Select New File'. If you have a better idea on how to handle this while maintaining the same styling, feel free to suggest it!